typegen/src/tests/mod.rs

```diff
@@ -0,0 +1,1143 @@
+            #[codec(crate = ::subxt_path::ext::codec)]
+            #[decode_as_type(crate_path = ":: subxt_path :: ext :: scale_decode")]
+            #[encode_as_type(crate_path = ":: subxt_path :: ext :: scale_encode")]
+            pub struct Parent {
+                pub a: ::core::primitive::bool,
+                pub b: root::scale_typegen::tests::Child,
+            }
+        }
+    };
+
+    assert_eq!(code.to_string(), expected_code.to_string());
+}
+
+#[test]
+fn generate_tuple_struct() {
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct Parent(bool, Child);
+
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct Child(i32);
+
+    let code = Testgen::new()
+        .with::<Parent>()
+        .gen_tests_mod(subxt_settings());
+
+    let expected_code = quote! {
+        pub mod tests {
+            use super::root;
+
+            #[derive(::subxt_path::ext::codec::Decode, ::subxt_path::ext::codec::Encode, ::subxt_path::ext::scale_decode::DecodeAsType, ::subxt_path::ext::scale_encode::EncodeAsType, Debug)]
+            #[codec(crate = ::subxt_path::ext::codec)]
+            #[decode_as_type(crate_path = ":: subxt_path :: ext :: scale_decode")]
+            #[encode_as_type(crate_path = ":: subxt_path :: ext :: scale_encode")]
+            pub struct Child(pub ::core::primitive::i32,);
+
+            #[derive(::subxt_path::ext::codec::Decode, ::subxt_path::ext::codec::Encode, ::subxt_path::ext::scale_decode::DecodeAsType, ::subxt_path::ext::scale_encode::EncodeAsType, Debug)]
+            #[codec(crate = ::subxt_path::ext::codec)]
+            #[decode_as_type(crate_path = ":: subxt_path :: ext :: scale_decode")]
+            #[encode_as_type(crate_path = ":: subxt_path :: ext :: scale_encode")]
+            pub struct Parent(pub ::core::primitive::bool, pub root::scale_typegen::tests::Child,);
+        }
+    };
+
+    assert_eq!(code.to_string(), expected_code.to_string());
+}
+
+#[test]
+fn derive_compact_as_for_uint_wrapper_structs() {
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct Su8 {
+        a: u8,
+    }
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct TSu8(u8);
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct Su16 {
+        a: u16,
+    }
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct TSu16(u16);
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct Su32 {
+        a: u32,
+    }
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct TSu32(u32);
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct Su64 {
+        a: u64,
+    }
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct TSu64(u64);
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct Su128 {
+        a: u128,
+    }
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct TSu128(u128);
+
+    let code = Testgen::new()
+        .with::<Su8>()
```

Comment:
super small nit: Maybe `register_type` instead of `with`, should be ok either way since its a testing mod